osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -71,6 +74,12 @@ public DrawableSpinner(Spinner s) : base(s)
             disc.Scale = scaleToCircle;
         }
 
+        [BackgroundDependencyLoader(permitNulls: true)]
+        private void load(OsuGame game)
+        {
+            beatmapOd = game?.Beatmap?.Value.Beatmap.BeatmapInfo.Difficulty.OverallDifficulty ?? 5;

```

Comment:
You shouldn't be retrieving spins required like this. Override ApplyDefaults() in Spinner.cs and set a publicly visible local "SpinsRequired" or something, then refer to it in DrawableSpinner by `HitObject.SpinsRequired`. Look at DrumRoll.cs for an example of this.